Redis integration tests have side effects

### How do you use Sentry?

Self-hosted/on-premise

### Version

1.9.2

### Steps to Reproduce

While working on https://github.com/getsentry/sentry-python/pull/1543, I noticed the following:

1. Checked out `sentry-sdk` for development.
2. Installed redis:
    ```
    fakeredis==1.9.0
    redis==3.5.3
    redis-py-cluster==2.1.3
    ````
3. Run redis integration tests twice, in different order:
    ```bash
    # first rediscluster, then redis
    pytest 'tests/integrations/rediscluster/test_rediscluster.py::test_rediscluster_basic[RedisCluster]' tests/integrations/redis/test_redis.py::test_basic
    # first redis, then rediscluster
    pytest tests/integrations/redis/test_redis.py::test_basic 'tests/integrations/rediscluster/test_rediscluster.py::test_rediscluster_basic[RedisCluster]'

### Expected Result

Both test runs pass.

### Actual Result

The second test run

```bash
pytest tests/integrations/redis/test_redis.py::test_basic 'tests/integrations/rediscluster/test_rediscluster.py::test_rediscluster_basic[RedisCluster]' 
```

fails with 

```pytest
tests/integrations/redis/test_redis.py .                                                                                                                                                                                                                                                    [ 50%]
tests/integrations/rediscluster/test_rediscluster.py F                                                                                                                                                                                                                                      [100%]

============================================================================================================================================ FAILURES =============================================================================================================================================
______________________________________________________________________________________________________________________________ test_rediscluster_basic[RedisCluster] ______________________________________________________________________________________________________________________________
tests/integrations/rediscluster/test_rediscluster.py:29: in test_rediscluster_basic
    (crumb,) = event["breadcrumbs"]["values"]
E   ValueError: not enough values to unpack (expected 1, got 0)
```

--- a/sentry_sdk/integrations/redis.py
+++ b/sentry_sdk/integrations/redis.py
@@ -131,7 +131,6 @@ def patch_redis_client(cls, is_cluster):
     This function can be used to instrument custom redis client classes or
     subclasses.
     """
-
     old_execute_command = cls.execute_command
 
     def sentry_patched_execute_command(self, name, *args, **kwargs):

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -19,6 +19,7 @@ from sentry_sdk._compat import reraise, string_types, iteritems
 from sentry_sdk.transport import Transport
 from sentry_sdk.envelope import Envelope
 from sentry_sdk.utils import capture_internal_exceptions
+from sentry_sdk.integrations import _installed_integrations  # noqa: F401
 
 from tests import _warning_recorder, _warning_recorder_mgr
 
@@ -163,6 +164,17 @@ def validate_event_schema(tmpdir):
             jsonschema.validate(instance=event, schema=SENTRY_EVENT_SCHEMA)
 
     return inner
+
+
+@pytest.fixture
+def reset_integrations():
+    """
+    Use with caution, sometimes we really need to start
+    with a clean slate to ensure monkeypatching works well,
+    but this also means some other stuff will be monkeypatched twice.
+    """
+    global _installed_integrations
+    _installed_integrations.clear()
 
 
 @pytest.fixture

--- a/tests/integrations/rediscluster/test_rediscluster.py
+++ b/tests/integrations/rediscluster/test_rediscluster.py
@@ -11,8 +11,8 @@ if hasattr(rediscluster, "StrictRedisCluster"):
     rediscluster_classes.append(rediscluster.StrictRedisCluster)
 
 
-@pytest.fixture(scope="module", autouse=True)
-def monkeypatch_rediscluster_classes():
+@pytest.fixture(autouse=True)
+def monkeypatch_rediscluster_classes(reset_integrations):
 
     try:
         pipeline_cls = rediscluster.ClusterPipeline

--- a/tests/integrations/sanic/test_sanic.py
+++ b/tests/integrations/sanic/test_sanic.py
@@ -1,5 +1,5 @@
+import os
 import sys
-
 import random
 import asyncio
 from unittest.mock import Mock
@@ -18,6 +18,20 @@ SANIC_VERSION = tuple(map(int, SANIC_VERSION_RAW.split(".")))
 
 @pytest.fixture
 def app():
+    if SANIC_VERSION < (19,):
+        """
+        Older Sanic versions 0.8 and 18 bind to the same fixed port which
+        creates problems when we run tests concurrently.
+        """
+        old_test_client = Sanic.test_client.__get__
+
+        def new_test_client(self):
+            client = old_test_client(self, Sanic)
+            client.port += os.getpid() % 100
+            return client
+
+        Sanic.test_client = property(new_test_client)
+
     if SANIC_VERSION >= (20, 12):
         # Build (20.12.0) adds a feature where the instance is stored in an internal class
         # registry for later retrieval, and so add register=False to disable that

